test/src/test/java/org/corfudb/runtime/collections/CorfuStoreShimTest.java

```diff
@@ -862,6 +863,171 @@ public void testNestedSecondaryIndexesRepeatedFieldFollowedByNonPrimitive() thro
+                .addContacts(contactsWithPhone.get(0)) // Common phone number (work and family)
+                .addContacts(contactsWithPhone.get(index)) // Unique work number with common mobile
+                .build();
+
+        // (2) Contact Book for Friends
+        index++;
+        Uuid contactBookFriendsId = getUuid();
+        ExampleSchemas.ContactBook contactBookFriends = ExampleSchemas.ContactBook.newBuilder()
+                .setId(ExampleSchemas.ContactBookId.newBuilder().setUuid(contactBookFriendsId.toString()).build())
+                .setApplication("iPhone")
+                .addContacts(contactsWithAddress.get(index - 1)) // San Francisco Address
+                .addContacts(contactsWithPhone.get(index)) // Unique friends number with common mobile
+                .build();
+
+        // (3) Contact Book for Family
+        index++;
+        Uuid contactBookFamilyId = getUuid();
+        ExampleSchemas.ContactBook contactBookFamily = ExampleSchemas.ContactBook.newBuilder()
+                .setId(ExampleSchemas.ContactBookId.newBuilder().setUuid(contactBookFamilyId.toString()).build())
+                .setApplication("iPhone")
+                .addContacts(contactsWithAddress.get(index - 1)) // Los Angeles Address
+                .addContacts(contactsWithPhone.get(index)) // Unique family number with common mobile
+                .addContacts(contactsWithPhone.get(++index)) // Unique family with unique phone number
+                .addContacts(contactsWithPhone.get(++index)) // Common phone number (work and family)
+                .build();
+
+        // Add Contact Books (work, friends, family)
+        try (ManagedTxnContext txn = shimStore.tx(someNamespace)) {
+            txn.putRecord(tableName, contactBookWorkId, contactBookWork, user);
+            txn.putRecord(tableName, contactBookFriendsId, contactBookFriends, user);
+            txn.putRecord(tableName, contactBookFamilyId, contactBookFamily, user);
+            txn.commit();
+        }
+
+        List<Uuid> contactBooksWithAddressSFO = Arrays.asList(contactBookWorkId, contactBookFriendsId);
+        List<Uuid> contactBooksWithCommonNumber = Arrays.asList(contactBookWorkId, contactBookFamilyId);
+        List<Uuid> contactBooksWithCommonMobile = Arrays.asList(contactBookWorkId, contactBookFriendsId, contactBookFamilyId);
+        List<Uuid> contactBooksIPhone = Arrays.asList(contactBookFriendsId, contactBookFamilyId);
+
+
+        // Get by secondary index, retrieve number of contactBooks that have contacts with address in San Francisco
+        // only contactBookWork and contactBookFriends have contacts with address in SFO.
+        try (ManagedTxnContext readWriteTxn = shimStore.tx(someNamespace)) {
+            List<CorfuStoreEntry<Uuid, ExampleSchemas.ContactBook, ManagedMetadata>> contactBooks = readWriteTxn
+                    .getByIndex(table, "city", "San Francisco");
```

Comment:
dont we need to define the 'city' alias in the example protobuf to directly use it as the secondary index?

Reply:
It is defined, L.168 of the protobuf